testing/src/scenario/state.py

```diff
@@ -122,26 +122,40 @@ class _MaxPositionalArgs:
 
         _max_positional_args = n
 
-        def __new__(cls, *args: Any, **kwargs: Any):
+        @classmethod
+        def _annotate_class(cls):
+            """Record information about which parameters are positional vs. keyword-only."""
+            if hasattr(cls, "_init_parameters"):
+                # We don't support dynamically changing the signature of a
+                # class, so we assume here it's the same as previously.
```

Comment:
This function will only ever be used on our own classes, will it not?

Maybe it's better to state that instead.

Reply:
It's tricky - `_max_posargs()` is clearly marked as being only for our use, but then most of the state components are public and you could, theoretically, mess about with them - I doubt a test would ever have reason to do this, but maybe some library built on top of Scenario might? I'll adjust the comment so that it covers both points.